state/runtime/evm/instructions.go

```diff
@@ -842,6 +843,14 @@ func opReturnDataCopy(c *state) {
 	}
 
 	data := c.returnData[dataOffset.Uint64():dataEndIndex]
+	// this should not happen because the memory is allocated considering both memOffset and data length and
+	// in the case of length==0 the fn returns immediately, not reaching this point
+	if uint64(len(c.memory)) < memOffset.Uint64()+uint64(len(data)) {
+		c.exit(errReturnDataOutOfBounds)
+
+		return
+	}
+
```

Comment:
If this code path is never reached why are we checking it? how are we going to test it?

Reply:
I agree that this is not needed, because if the execution got to this point it means that size is greater then 0 and allocate fn allocated enough memory taking into consideration both memOffset and size. 
I'll remove this check.